preprocess/analysis/language_analysis.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Jun 23 16:07:17 2020
+
+@author: robert
+"""
+
+# Imports the Google Cloud client library
+from google.cloud import language
+from google.cloud.language import enums
+from google.cloud.language import types
+
+import json
+
+# See readme about how to setup mykey.json
+CLIENT = language.LanguageServiceClient.from_service_account_json("mykey.json")
+ENCODING_TYPE = enums.EncodingType.UTF8
+
+
+def round_to_1(number):
+  """Rounds number to the nearest tenth. 
+
+  Cloud language api will automatically return floats to like 15 decimal 
+  places. It's convenient to just round down. 
+
+  Args: 
+    number: float or an int.
+  
+  Returns:
+    a float to tens precisions.
+  """
+  
+  return float(round(number, 1))
+
+def round_to_2(number): 
+  """Rounds number to the nearest hundreth.
+  
+  Args: 
+    number: float or an int.
+  
+  Returns:
+    a float to hundreds precisions.
+  """
+  
+  return float(round(number, 2))
+
+
+
+def analyze_sentiment(text):
+  """Applies sentiment analysis to an input string.
+  
+  Calls cloud language API once for sentiment analysis if input is a string. 
+  Rounds sentiment and magnitude to the nearest tenth for convenience. 
+
+  Args: 
+    text: an input string.
+  
+  Returns: 
+    A json string mapping score and magnitude keys to their corresponding
+    value.
+    
+  Raises:
+    TypeError: input is not a string.
+  """
+  
+  if not isinstance(text, str): 
+    raise TypeError("Input to analyze_sentiment isn't a string.")
```

Comment:
Include the error string here for clarification later

Reply:
Done.